experiment/simulation/js/animation-ring-counter.js

```diff
@@ -0,0 +1,548 @@
+        clockDisappear();
+        TEXTCLOCK[0].textContent = 0;
+        svg.appendChild(TEXTCLOCK[0]);
+        setCoordinates(17,504,TEXTCLOCK[0]);
+        fillColor(OBJECTS[1],"#eeeb22");
+        clearObservation();
+        clockVisible();
+
+        for(const dot of CLOCKDOT){
+            setter(TEXTCLOCK[0].textContent,dot);
+        }
+    }
+    else if (TEXTCLOCK[0].textContent !== "1" && timeline.progress() === 0) {
+        setClock();
+    }
+}
+function setClock() {
+    clockDisappear();
+    TEXTCLOCK[0].textContent = 1;
+    svg.appendChild(TEXTCLOCK[0]);
+    setCoordinates(17,504,TEXTCLOCK[0]);
+    fillColor(OBJECTS[1],"#29e");
+    clearObservation();
+    clockVisible();
+    for(const dot of CLOCKDOT){
+        setter(TEXTCLOCK[0].textContent,dot);
+    }
+    OBSERV.innerHTML = "Clock has Started";
+}
+
+function reboot() {
+    TEXTINPUT[0].textContent = 2;
+    TEXTCLOCK[0].textContent = 2;
+}
+
+function outputSetter() {
+    for(let index=0;index<TEXTOUTPUT.length;index++){
+        setter(TEXTOUTPUT[index].textContent, OBJECTS[index+2]);
+        setter(TEXTOUTPUT[index].textContent, CLOCKDOT[index]);
+    }
+}
+function display() {
+    OBSERV.innerHTML = "Simulation has finished. Press Restart to start again"
+}
+function setter(value, component) {
+    if (value === "1") {
+        unsetColor(component);
+    }
+    else if (value === "0") {
+        setColor(component);
+    }
+}
+function clockToZero() {
+    TEXTCLOCK[0].textContent = 0;
+    svg.appendChild(TEXTCLOCK[0]);
+    setCoordinates(17,504,TEXTCLOCK[0]);
+    fillColor(OBJECTS[1],"#eeeb22");
+    OBSERV.innerHTML = "Negative edge triggered change in output expected now";
+}
+function clockToOne() {
+    TEXTCLOCK[0].textContent = 1;
+    svg.appendChild(TEXTCLOCK[0]);
+    setCoordinates(17,504,TEXTCLOCK[0]);
+    fillColor(OBJECTS[1],"#29e");
+    OBSERV.innerHTML = "No change in output";
+}
+
+function changeSpeed(newSpeed){
+    if (TEXTCLOCK[0].textContent !== "2" && TEXTINPUT[0].textContent !== "2" && timeline.progress() !== 1) {
+        timeline.resume();
+        timeline.timeScale(newSpeed);
+        OBSERV.innerHTML = "2x speed";
+        decide = 1;
+        STATUS.innerHTML = "Pause";
+    }
+}
+function setSpeed(speed) {
+    if (circuitStarted !== 0) {
+        if (speed === "1") {
+            changeSpeed(1);
+        }
+        else if (speed === "2") {
+            changeSpeed(2);
+        }
+        else if (speed === "4") {
+            changeSpeed(4);
+        }
+    }
+
+}
+
+function restartCircuit() {
+    if (circuitStarted === 1) {
+        circuitStarted = 0;
+    }
+    timeline.seek(0);
+    timeline.pause();
+    allDisappear();
+    reboot();
+    clearObservation();
+    decide = 0;
+    STATUS.innerHTML = "Start";
+    OBSERV.innerHTML = "Successfully restored";
+    SPEED.selectedIndex = 0;
+}
+
+function simulationStatus() {
+    if (decide === 0) {
```

Comment:
Any variables which are essentially use as flags for toggling on and off things, please prefer ti use a boolean type. It makes it more readable and makes the conditional statement concise

Reply:
changed to boolean